route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -43,188 +42,187 @@ namespace route_following_plugin
         plugin_discovery_msg_.capability = "strategic_plan/plan_maneuvers";
         pose_sub_ = nh_->subscribe("current_pose", 1, &RouteFollowingPlugin::pose_cb, this);
         twist_sub_ = nh_->subscribe("current_velocity", 1, &RouteFollowingPlugin::twist_cd, this);
-        
+
         pnh_->param<double>("minimal_maneuver_duration", mvr_duration_, 16.0);
-        pnh2_->param<double>("config_speed_limit",config_limit);
-        pnh_->param<double>("buffer_time_lanechange",buffer_lanechange_time_);
-        pnh2_->param<double>("guidance/route_end_jerk", jerk_, 1.0);
-        pnh2_->param<std::string>("guidance/lane_change_plugin", lane_change_plugin_);
+        pnh_->param<std::string>("lane_change_plugin", lane_change_plugin_);
+
         wml_.reset(new carma_wm::WMListener());
         // set world model point form wm listener
         wm_ = wml_->getWorldModel();
-
+        //set a route callback to update route and calculate maneuver
+        lanelet::BasicPoint2d current_loc(pose_msg_.pose.position.x, pose_msg_.pose.position.y);
+        wml_->setRouteCallback([this, &current_loc]() { this->latest_maneuver_plan_ = route_cb(wm_->getRoute()->shortestPath(), current_loc);});
```

Comment:
Don't try to capture a local variable which will go out of scope like this. The current_location variable, the way this is written, will never change even if rerouting occurs. Instead you want to extract it from "this" inside the lambda similar to what you did for latest_maneuver_plan_